docs/content/preview/benchmark/resilience/jepsen-testing.md

```diff
@@ -0,0 +1,210 @@
+UPDATE T SET balance = balance + x WHERE id = ?;
+DELETE FROM T WHERE id = ?;
+COMMIT;
+```
+
+### Counter
+
+This test uses a table T`(id int PRIMARY KEY, count int)` with a single row, with the workload
+consisting of concurrent increments `(UPDATE T SET count = count + ? WHERE id = 0)` and reads. At
+any
+given time the value column is expected to be not more than the number of increments issued, and not
+less than the number of increments succeeded.
+
+The test utilizes the int column type for YSQL and counter type for YCQL.
+
+![Load Phase Results](/images/benchmark/jepsen/jepsen-3-counter.png)
+
+### Set
+
+This scenario uses a table with T`(id int PRIMARY KEY, val int, grp int)`. Concurrently, values are
+inserted into this table while simultaneously reading the entire table. Once an insert operation
+succeeds, or when an element is first observed in a read operation (whichever occurs first), all
+subsequent reads are expected to include that inserted element.
+
+For YCQL, the table also includes a count column, and the workload permits duplicate inserts.
+
+![Load Phase Results](/images/benchmark/jepsen/jepsen-4-set.png)
+
+```plpgsql
+INSERT INTO T (id, val, grp) VALUES (?, ?, ?);
+SELECT val FROM T where id = 0;
+```
+
+### Long fork
+
+The long-fork test uses a table T`(key int PRIMARY KEY, key2 int, val int)`, where individual
+workers
+execute either single-row inserts or perform multi-row reads. The expectation is that the read
+results are serializable. This means that for two write operations, W1 and W2, it should not be
+possible for a read operation R1 to observe write W1 but not W2, while another read operation R2
+observes W2 but not W1.
+To read more about the long-fork test see [here](https://jepsen-io.github.io/jepsen/jepsen.tests.long-fork.html).
+
+![Load Phase Results](/images/benchmark/jepsen/jepsen-5-long-fork.png)
+
+### Default value
+
+This scenario entails concurrent Data Definition Language (DDL) and Data Manipulation Language (DML)
+operations, simulating a migration scenario. Here, the user adds a `DEFAULT 0` column, ensuring that
+it is actually zero and not null when performing inserts and reads.
+
+![Load Phase Results](/images/benchmark/jepsen/jepsen-6-default-value.png)
+
+### Single Key ACID
+
+The test uses a table T`(id int PRIMARY KEY, val int)` with a fixed number of rows, each row having
+several worker threads assigned to it. Each worker can either read the row, update the row, or
+perform compare-and-set `UPDATE T SET val = ? WHERE id = ? AND val = ?`, worker groups for different
+rows are completely independent of each other. Checker makes sure that the resulting operations
+history is linearizable - i.e. that reads observe previous writes and writes don’t disappear.
+
+![Load Phase Results](/images/benchmark/jepsen/jepsen-7-single-key-acid.png)
+
+### Multi Key ACID
+
+Similar to the single-key ACID test, but uses a composite primary key T`(k1 int, k2 int, val int,
```

Comment:
```suggestion
This test is similar to the single-key ACID test, but uses a composite primary key on table T with schema `(k1 int, k2 int, val int, PRIMARY KEY (k1, k2))` and UPSERTs (for YSQL it is `INSERT .. ON CONFLICT DO UPDATE SET`) instead of
compare-and-set.
```

Reply:
Fixed